### ISSUE TITLE
Contest judging system

### ISSUE THREAD

**Issue by Ephemeralis** [ppy/osu-web#903]
The contest system should have the following features for proper judge-based voting:

* Support for up to 16 judges
* Support for double-blind voting by masking entrant usernames with a masked name, including altering .osu file metadata to reflect this masked name
* Support for up to 12 custom judging score categories on a 1-10 rating scale. Score categories are things like "flow and structure", or "judge's impression", though ideally provision would exist to create these categories per contest instead of being limited to a certain set (as per [the old system](https://puu.sh/t8yJE/e96dce870f.png))
* Per entry, a text field for judge comments
* Results page per user entry including judging comments and per judge scoring in the various aforementioned score categories
* Contest ranking page that displays winning entries and all other entries sorted by score
* Panel which displays overall judging completion percent (visible only to administrators)

Basically if /p/contestresults and /p/contest have it, it should be in the new contest system as well.
